Issues when running tests

I'm facing some issues when running the tests:

Command: `$ cargo test --all-features `

Output:
```
warning: profile package spec `blas-tests` in profile `test` did not match any packages
warning: profile package spec `numeric-tests` in profile `test` did not match any packages
   Compiling ndarray v0.16.1 (/mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray)
error[E0433]: failed to resolve: use of undeclared crate or module `ndarray_gen`
  --> tests/oper.rs:13:5
   |
13 | use ndarray_gen::array_builder::ArrayBuilder;
   |     ^^^^^^^^^^^ use of undeclared crate or module `ndarray_gen`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `ndarray` (test "oper") due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: linking with `cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/usr/lib/rustlib/x86_64-unknown-linux-gnu/bin:/usr/lib/lib/rustlib/x86_64-unknown-linux-gnu/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games" VSLANG="1033" "cc" "-m64" "/tmp/rustc257VgV/symbols.o" "<170 object files omitted>" "-Wl,--as-needed" "-Wl,-Bstatic" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/{libtest-513130a04fdaef45.rlib,libgetopts-6694e698be18b90e.rlib,libunicode_width-54f69923c57f31d4.rlib,librustc_std_workspace_std-faafef83820e78c4.rlib}" "/mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/target/debug/deps/{libndarray-98b3905d2e210e9d.rlib,libmatrixmultiply-f5e0b8dc49ca2879.rlib,libnum_cpus-8582989cae2f4e1a.rlib,libthread_tree-5a98188572a8c7fe.rlib,libcrossbeam_channel-65212d96a1600ae0.rlib,libonce_cell-2c106065c6ce175f.rlib,libnum_complex-48bea12be644d291.rlib,librayon-b267a8950c95e36f.rlib,librayon_core-4c066a6854f1f895.rlib,libcrossbeam_deque-ed1e616185f2ca71.rlib,libcrossbeam_epoch-425a36f50195b3d8.rlib,libcrossbeam_utils-08eb64f27b325744.rlib,libnum_integer-27c7a7cf710ecf75.rlib,librawpointer-a51917bf0f0a91f6.rlib,libserde-5238e1621d7b6d0e.rlib,libcblas_sys-1b4c68c5a18fd883.rlib,liblibc-4b73429492b96d7d.rlib,libitertools-77cd7349df7df26f.rlib,libeither-289c4669589ef15b.rlib,libdefmac-79cae88a8d84e49b.rlib,libapprox-0392ab4571a8832d.rlib,libnum_traits-d28af6a134b7ba10.rlib}" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/{libstd-d0233f198e8667ce.rlib,libpanic_unwind-21c2e105ea31789b.rlib,libobject-8c6eb1e7f3610c33.rlib,libmemchr-a47865c8f19aabd9.rlib,libaddr2line-372dcfa705f88375.rlib,libgimli-15a40a9df6fe2af6.rlib,librustc_demangle-8d104450181f07c5.rlib,libstd_detect-d14dc0ea762689ce.rlib,libhashbrown-da3dfe6fe9c04e44.rlib,librustc_std_workspace_alloc-61c7bed2505d3c86.rlib,libminiz_oxide-ac1cc67970c69c19.rlib,libadler-49d4372422a1c03e.rlib,libunwind-de4de4fa039bbc1b.rlib,libcfg_if-d400e76a97db55fa.rlib,liblibc-e013e3e11405069d.rlib,liballoc-fe77680e832d62e1.rlib,librustc_std_workspace_core-a4256da78119cf51.rlib,libcore-ed1ff47b6d506937.rlib,libcompiler_builtins-cc3a99197fff52a0.rlib}" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/target/debug/deps/array-861fb9eaf211d419" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs"
  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: /usr/bin/ld: /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/target/debug/deps/array-861fb9eaf211d419.0xicy0dv9tk24gfkpik0zufu1.rcgu.o: in function `ndarray::linalg::impl_linalg::<impl ndarray::ArrayBase<S,ndarray::dimension::dim::Dim<[usize; 1]>>>::dot_impl':
          /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/src/linalg/impl_linalg.rs:118:(.text._ZN7ndarray6linalg11impl_linalg110_$LT$impl$u20$ndarray..ArrayBase$LT$S$C$ndarray..dimension..dim..Dim$LT$$u5b$usize$u3b$$u20$1$u5d$$GT$$GT$$GT$8dot_impl17h1d58c52ddb8159b1E+0x343): undefined reference to `cblas_ddot'
          /usr/bin/ld: /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/src/linalg/impl_linalg.rs:118:(.text._ZN7ndarray6linalg11impl_linalg110_$LT$impl$u20$ndarray..ArrayBase$LT$S$C$ndarray..dimension..dim..Dim$LT$$u5b$usize$u3b$$u20$1$u5d$$GT$$GT$$GT$8dot_impl17h1d58c52ddb8159b1E+0x493): undefined reference to `cblas_sdot'
          /usr/bin/ld: /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/target/debug/deps/array-861fb9eaf211d419.a74x1flyg7d0cvksfozy99u7u.rcgu.o: in function `ndarray::linalg::impl_linalg::mat_mul_impl':
          /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/src/linalg/impl_linalg.rs:427:(.text._ZN7ndarray6linalg11impl_linalg12mat_mul_impl17h796b8d27ba9832e8E+0x60c): undefined reference to `cblas_sgemm'
          /usr/bin/ld: /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/src/linalg/impl_linalg.rs:427:(.text._ZN7ndarray6linalg11impl_linalg12mat_mul_impl17h796b8d27ba9832e8E+0x734): undefined reference to `cblas_dgemm'
          /usr/bin/ld: /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/src/linalg/impl_linalg.rs:427:(.text._ZN7ndarray6linalg11impl_linalg12mat_mul_impl17h796b8d27ba9832e8E+0x831): undefined reference to `cblas_cgemm'
          /usr/bin/ld: /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/src/linalg/impl_linalg.rs:427:(.text._ZN7ndarray6linalg11impl_linalg12mat_mul_impl17h796b8d27ba9832e8E+0x8fe): undefined reference to `cblas_zgemm'
          /usr/bin/ld: /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/target/debug/deps/array-861fb9eaf211d419.a74x1flyg7d0cvksfozy99u7u.rcgu.o: in function `ndarray::linalg::impl_linalg::mat_mul_impl':
          /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/src/linalg/impl_linalg.rs:427:(.text._ZN7ndarray6linalg11impl_linalg12mat_mul_impl17hfceefcb57efc1743E+0x608): undefined reference to `cblas_sgemm'
          /usr/bin/ld: /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/src/linalg/impl_linalg.rs:427:(.text._ZN7ndarray6linalg11impl_linalg12mat_mul_impl17hfceefcb57efc1743E+0x730): undefined reference to `cblas_dgemm'
          /usr/bin/ld: /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/src/linalg/impl_linalg.rs:427:(.text._ZN7ndarray6linalg11impl_linalg12mat_mul_impl17hfceefcb57efc1743E+0x82d): undefined reference to `cblas_cgemm'
          /usr/bin/ld: /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/src/linalg/impl_linalg.rs:427:(.text._ZN7ndarray6linalg11impl_linalg12mat_mul_impl17hfceefcb57efc1743E+0x8fa): undefined reference to `cblas_zgemm'
          /usr/bin/ld: /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/target/debug/deps/array-861fb9eaf211d419.a74x1flyg7d0cvksfozy99u7u.rcgu.o: in function `ndarray::linalg::impl_linalg::general_mat_vec_mul_impl':
          /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/src/linalg/impl_linalg.rs:680:(.text._ZN7ndarray6linalg11impl_linalg24general_mat_vec_mul_impl17h516343ddbded6d7dE+0x4ec): undefined reference to `cblas_sgemv'
          /usr/bin/ld: /mnt/packages/git_repos/dpkg/debcargo-conf/build/ndarray/src/linalg/impl_linalg.rs:680:(.text._ZN7ndarray6linalg11impl_linalg24general_mat_vec_mul_impl17h516343ddbded6d7dE+0x7f2): undefined reference to `cblas_dgemv'
          collect2: error: ld returned 1 exit status
          
  = note: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#rustc-link-lib)

error: could not compile `ndarray` (test "array") due to 1 previous error
```

-----------

Command: `cargo test --all-targets --no-default-features --features approx`

```
 
 error[E0599]: no function or associated item named `linspace` found for struct `ArrayBase` in the current scope
   --> tests/azip.rs:235:22
    |
235 |     let res = Array::linspace(0., 3.1, 32).mapv_into(f32::sin);
    |                      ^^^^^^^^ function or associated item not found in `ArrayBase<OwnedRepr<_>, _>`
    |
note: if you're trying to build a new `ArrayBase<OwnedRepr<_>, _>` consider using one of the following associated functions:
      ndarray::impl_constructors::<impl ArrayBase<S, Dim<[usize; 1]>>>::from_vec
      ndarray::impl_constructors::<impl ArrayBase<S, Dim<[usize; 1]>>>::from_iter
      ndarray::impl_constructors::<impl ArrayBase<S, Dim<[usize; 2]>>>::eye
      ndarray::impl_constructors::<impl ArrayBase<S, Dim<[usize; 2]>>>::from_diag
      and 9 others
   --> /usr/share/cargo/registry/ndarray-0.16.1/src/impl_constructors.rs:58:5
```

-----------


Command: `cargo test --all-targets --no-default-features --features blas`

```
error: cannot find macro `eprintln` in this scope
    --> src/linalg/impl_linalg.rs:1060:13
     |
1060 |             eprintln!("{:?}", m);
     |             ^^^^^^^^
     |
help: consider importing this macro
     |
958  +     use std::eprintln;
     |
```

--- a/benches/bench1.rs
+++ b/benches/bench1.rs
@@ -982,6 +982,7 @@ fn dot_extended(bench: &mut test::Bencher)
 
 const MEAN_SUM_N: usize = 127;
 
+#[cfg(feature = "std")]
 fn range_mat(m: Ix, n: Ix) -> Array2<f32>
 {
     assert!(m * n != 0);
@@ -990,6 +991,7 @@ fn range_mat(m: Ix, n: Ix) -> Array2<f32>
         .unwrap()
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn mean_axis0(bench: &mut test::Bencher)
 {
@@ -997,6 +999,7 @@ fn mean_axis0(bench: &mut test::Bencher)
     bench.iter(|| a.mean_axis(Axis(0)));
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn mean_axis1(bench: &mut test::Bencher)
 {
@@ -1004,6 +1007,7 @@ fn mean_axis1(bench: &mut test::Bencher)
     bench.iter(|| a.mean_axis(Axis(1)));
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn sum_axis0(bench: &mut test::Bencher)
 {
@@ -1011,6 +1015,7 @@ fn sum_axis0(bench: &mut test::Bencher)
     bench.iter(|| a.sum_axis(Axis(0)));
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn sum_axis1(bench: &mut test::Bencher)
 {

--- a/benches/construct.rs
+++ b/benches/construct.rs
@@ -19,6 +19,7 @@ fn zeros_f64(bench: &mut Bencher)
     bench.iter(|| Array::<f64, _>::zeros((128, 128)))
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn map_regular(bench: &mut test::Bencher)
 {
@@ -28,6 +29,7 @@ fn map_regular(bench: &mut test::Bencher)
     bench.iter(|| a.map(|&x| 2. * x));
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn map_stride(bench: &mut test::Bencher)
 {

--- a/benches/higher-order.rs
+++ b/benches/higher-order.rs
@@ -12,6 +12,7 @@ const N: usize = 1024;
 const X: usize = 64;
 const Y: usize = 16;
 
+#[cfg(feature = "std")]
 #[bench]
 fn map_regular(bench: &mut Bencher)
 {
@@ -26,6 +27,7 @@ pub fn double_array(mut a: ArrayViewMut2<'_, f64>)
     a *= 2.0;
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn map_stride_double_f64(bench: &mut Bencher)
 {
@@ -38,6 +40,7 @@ fn map_stride_double_f64(bench: &mut Bencher)
     });
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn map_stride_f64(bench: &mut Bencher)
 {
@@ -48,6 +51,7 @@ fn map_stride_f64(bench: &mut Bencher)
     bench.iter(|| av.map(|&x| 2. * x));
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn map_stride_u32(bench: &mut Bencher)
 {
@@ -59,6 +63,7 @@ fn map_stride_u32(bench: &mut Bencher)
     bench.iter(|| av.map(|&x| 2 * x));
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn fold_axis(bench: &mut Bencher)
 {

--- a/benches/iter.rs
+++ b/benches/iter.rs
@@ -45,6 +45,7 @@ fn iter_sum_2d_transpose(bench: &mut Bencher)
     bench.iter(|| a.iter().sum::<i32>());
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn iter_filter_sum_2d_u32(bench: &mut Bencher)
 {
@@ -55,6 +56,7 @@ fn iter_filter_sum_2d_u32(bench: &mut Bencher)
     bench.iter(|| b.iter().filter(|&&x| x < 75).sum::<u32>());
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn iter_filter_sum_2d_f32(bench: &mut Bencher)
 {
@@ -65,6 +67,7 @@ fn iter_filter_sum_2d_f32(bench: &mut Bencher)
     bench.iter(|| b.iter().filter(|&&x| x < 75.).sum::<f32>());
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn iter_filter_sum_2d_stride_u32(bench: &mut Bencher)
 {
@@ -76,6 +79,7 @@ fn iter_filter_sum_2d_stride_u32(bench: &mut Bencher)
     bench.iter(|| b.iter().filter(|&&x| x < 75).sum::<u32>());
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn iter_filter_sum_2d_stride_f32(bench: &mut Bencher)
 {
@@ -87,6 +91,7 @@ fn iter_filter_sum_2d_stride_f32(bench: &mut Bencher)
     bench.iter(|| b.iter().filter(|&&x| x < 75.).sum::<f32>());
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn iter_rev_step_by_contiguous(bench: &mut Bencher)
 {
@@ -98,6 +103,7 @@ fn iter_rev_step_by_contiguous(bench: &mut Bencher)
     });
 }
 
+#[cfg(feature = "std")]
 #[bench]
 fn iter_rev_step_by_discontiguous(bench: &mut Bencher)
 {

--- a/benches/numeric.rs
+++ b/benches/numeric.rs
@@ -9,6 +9,7 @@ const N: usize = 1024;
 const X: usize = 64;
 const Y: usize = 16;
 
+#[cfg(feature = "std")]
 #[bench]
 fn clip(bench: &mut Bencher)
 {

--- a/src/linalg/impl_linalg.rs
+++ b/src/linalg/impl_linalg.rs
@@ -1071,7 +1071,6 @@ mod blas_tests
 
         for stride in 1..=MAXSTRIDE {
             let m = ArrayView::from_shape((N, N).strides((stride, 1)), &data).unwrap();
-            eprintln!("{:?}", m);
 
             if stride < N {
                 assert_eq!(get_blas_compatible_layout(&m), None);

